data-structures/iarray/IArrays.scala

```diff
@@ -36,6 +36,95 @@ object IArrays:
       res
     }.ensuring(_ == IArray(list.islice(from, until)))
 
+    @pure @extern
+    def concat(other: IArray[T]): IArray[T] = {
+      @ghost val list = this.list ++ other.list
+      val res = IArray(list)
+      val newArr = new Array[AnyRef](this.size + other.size)
+      this._arr.copyToArray(newArr, this._offset, this._offset + this._size)
+      other._arr.copyToArray(newArr, other._offset, other._offset + other._size)
```

Comment:
What is a documentation link for copyToArray? I don't know which paramaeter does what here.